src/lib/swapper/swappers/OsmosisSwapper/endpoints.ts

```diff
@@ -155,29 +160,65 @@ export const osmosisApi: Swapper2Api = {
 
   checkTradeStatus: async ({
     txHash,
-    chainId,
+    quoteId,
+    stepIndex,
+    quoteSellAssetAccountId,
+    quoteBuyAssetAccountId,
+    getState,
   }): Promise<{ status: TxStatus; buyTxHash: string | undefined; message: string | undefined }> => {
     try {
-      const { REACT_APP_OSMOSIS_NODE_URL: osmoUrl, REACT_APP_COSMOS_NODE_URL: cosmosUrl } =
-        getConfig()
-      assertGetCosmosSdkChainAdapter(chainId)
-
-      const status = await (async () => {
-        const baseUrl = chainId === osmosisChainId ? osmoUrl : cosmosUrl
-        const txResponse = await axios.get(`${baseUrl}/lcd/txs/${txHash}`)
-
-        if (!txResponse?.data) return TxStatus.Pending
-
-        if (!txResponse?.data?.codespace && !!txResponse?.data?.gas_used) return TxStatus.Confirmed
-        if (txResponse?.data?.codespace) return TxStatus.Failed
-
-        return TxStatus.Pending
-      })()
-
-      return {
-        status,
-        buyTxHash: txHash,
-        message: undefined,
+      const quote = tradeQuoteMetadata.get(quoteId)
+      const step = quote?.steps[stepIndex]
+      if (!step) throw new Error('Step not found')
+      const isAtomOsmoQuote =
+        step.sellAsset.chainId === cosmosChainId && step.buyAsset.chainId === osmosisChainId
+      const isIbcTransferStep = step.buyAsset.chainId !== step.sellAsset.chainId
+      if (!(quoteSellAssetAccountId && quoteBuyAssetAccountId))
+        throw new Error('quote AccountIds required to check osmosis trade status')
+      if (isIbcTransferStep) {
+        const stepSellAssetAccountId = isAtomOsmoQuote
+          ? quoteSellAssetAccountId
+          : quoteBuyAssetAccountId
+        const initiatingChainTxid = serializeTxIndex(
+          stepSellAssetAccountId,
+          txHash,
+          fromAccountId(stepSellAssetAccountId).account,
+        )
+        const pollResult = await pollForCrossChainComplete({
+          initiatingChainTxid,
+          initiatingChainAccountId: stepSellAssetAccountId,
+          getState,
+        })
+        const status = pollResult === 'success' ? TxStatus.Confirmed : TxStatus.Failed
```

Comment:
I've had success on the ATOM -> OSMO first hop step status here, unfortunately the node gave up before I had a chance to test the second branch. Since Osmosis flag is off at the moment, I'd say we get this in as-is for now and test it once nodes are happy before turning the flag back on (or maybe keep it off? 👋🏽 )

Reply:
I was able to get both directions working successfully on the production node this evening 👌